packages/core/src/change_detection/scheduling/zoneless_scheduling_impl.ts

```diff
@@ -176,7 +179,11 @@ export class ChangeDetectionSchedulerImpl implements ChangeDetectionScheduler {
     }
     // If we're inside the zone don't bother with scheduler. Zone will stabilize
     // eventually and run change detection.
-    if (!this.zonelessEnabled && this.zoneIsDefined && NgZone.isInAngularZone()) {
+    if (
+      !this.zonelessEnabled &&
+      this.zoneIsDefined &&
+      Zone.current.get(angularZoneInstanceIdProperty + this.angularZoneId)
```

Comment:
Wouldn't it make sense to compare the instance id prop itself, avoiding the need for the third property?

```suggestion
      Zone.current.get(angularZoneInstanceIdProperty) === this.angularZoneId
```

Reply:
This doesn't quite work for two reasons:

* You could be in a child zone of the Angular zone so it might not be the exact zone you're in
* There can be two apps on the page, and one might even live within the other. The presence of a property in the hierarchy doesn't mean it's the app's Zone that you're looking for.

Reply:
Oooh, that is very subtle but good points!

Reply:
I meant to write a test for these cases. Let me see what I can do.